### ISSUE TITLE
[Bug]: L0 segment related incorrect monitoring metrics

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#29204]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. invalid datacoord segment num on image: master-20231213-fe1eeae2
 - growing segment num should not be negative
 - there should be no segments of legacy segment type
![image](https://github.com/milvus-io/milvus/assets/27288593/619e4b4d-cae6-473f-9893-779d63ad0909)

2. same issue on querynode Segment loaded num
 - there should be no segments of legacy segment type
![image](https://github.com/milvus-io/milvus/assets/27288593/800a1a82-67ff-46b0-a2ca-dff163192724)

3. some metrics have no data. I guess It was not reported under some conditions ?
- datanode unflush segment num metric no data
- what's the relationship between <Autoflush operator rate>, <Flush operator rate>, < Flush request rate> ?
   If <Flush operator rate> represents manual and auto flush, then in the picture bellow, why are the manual operations successful but the <Flush operator rate> metrics no data ?
![image](https://github.com/milvus-io/milvus/assets/27288593/fa85bfdf-e746-4f29-aee8-b57734b256d3)
(https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=delta-par-key5&var-collection=All&var-app_name=milvus&from=1702466852008&to=1702489011899)

**Comment by XuanYang-cn** [comment]
#29190 fix the lagecy-segment bug

**Comment by XuanYang-cn** [comment]
#29194 fix the negative growing-l0 bug

**Comment by ThreadDao** [comment]
@XuanYang-cn  negative Dropped-L1 segments and wrong Flushed-L1 segments
![image](https://github.com/milvus-io/milvus/assets/27288593/9b7d263d-a7a0-4164-82a5-c2785efc4536)

- image: milvus-io-master-c0f0548-20240116
- argo: [delta-stable-2-8s82m-782514036](https://argo-workflows.zilliz.cc/archived-workflows/qa/2a05c2ea-de13-4739-bf4b-cbf5dff12549?nodeId=delta-stable-2-8s82m-782514036)
- grafana: [grafana of level-zero-multi-par-8a](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=level-zero-multi-par-8a&var-collection=All&var-app_name=milvus&from=1705407000000&to=1705468200000)
- birdwatcher segments
![image](https://github.com/milvus-io/milvus/assets/27288593/0dde76a6-de91-4282-b5b1-f2e70f1ebee3)
- queryNode loaded segments
![image](https://github.com/milvus-io/milvus/assets/27288593/3bf76c09-7301-4d80-9e85-3cef9e89986d)
- test steps:
   1. insert 3m into 3 partition (_default, partition_1, partition_2) and each partition has 1m
   2. index and load
   3. concurrent search + query + delete: id > 3m from _default partition + insert: insert data into a random partition, total 10 partitions
```
[{"name":"_default","collection_name":"fouram_w3K6u70b","description":""}, {"name":"partition_1","collection_name":"fouram_w3K6u70b","description":""}, {"name":"partition_2","collection_name":"fouram_w3K6u70b","description":""}, {"name":"p_JBPY452l","collection_name":"fouram_w3K6u70b","description":""}, {"name":"p_tAgitK60","collection_name":"fouram_w3K6u70b","description":""}, {"name":"p_ZLXrQm8z","collection_name":"fouram_w3K6u70b","description":""}, {"name":"p_4oSH895T","collection_name":"fouram_w3K6u70b","description":""}, {"name":"p_5bEvTYUU","collection_name":"fouram_w3K6u70b","description":""}, {"name":"p_2MrLWhzO","collection_name":"fouram_w3K6u70b","description":""}, {"name":"p_G3KJsXKW","collection_name":"fouram_w3K6u70b","description":""}]
```

- pods:
```
delta-multi-par-2-milvus-datanode-84f4d88696-56757                1/1     Running            0               18h     10.104.13.166   4am-node16   <none>           <none>
delta-multi-par-2-milvus-datanode-84f4d88696-b2qrl                1/1     Running            0               18h     10.104.33.124   4am-node36   <none>           <none>
delta-multi-par-2-milvus-indexnode-6b7954cc79-6lz5r               1/1     Running            0               18h     10.104.30.109   4am-node38   <none>           <none>
delta-multi-par-2-milvus-indexnode-6b7954cc79-rkk4n               1/1     Running            0               18h     10.104.19.212   4am-node28   <none>           <none>
delta-multi-par-2-milvus-mixcoord-674688bb7-qblrl                 1/1     Running            0               18h     10.104.30.108   4am-node38   <none>           <none>
delta-multi-par-2-milvus-proxy-8688fb854c-dkn6h                   1/1     Running            0               18h     10.104.31.184   4am-node34   <none>           <none>
delta-multi-par-2-milvus-querynode-884774864-2drmz                1/1     Running            0               18h     10.104.12.145   4am-node17   <none>           <none>
delta-multi-par-2-milvus-querynode-884774864-j9nw6                1/1     Running            0               18h     10.104.32.74    4am-node39   <none>           <none>
delta-multi-par-2-milvus-querynode-884774864-rjm8c                1/1     Running            0               18h     10.104.20.82    4am-node22   <none>           <none>

```

**Comment by ThreadDao** [comment]
@XuanYang-cn Legacy segment state in milvus_datanode_autoflush_buffer_op_count and milvus_datanode_flush_buffer_op_count
- image: milvus-io-master-c0f0548-20240116
![image](https://github.com/milvus-io/milvus/assets/27288593/13d21a97-d22c-4817-baed-f31e27eb8f1a)
 
![image](https://github.com/milvus-io/milvus/assets/27288593/ba51d5f8-a933-481e-9fe6-a8d8f957cb13)

- grafana: [grafana of delta-stable-dml-2](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=delta-stable-dml-2&var-collection=All&var-app_name=milvus&from=1705407000000&to=1705464600000)

- argo: [delta-stable-2-8s82m-1306261796](https://argo-workflows.zilliz.cc/archived-workflows/qa/2a05c2ea-de13-4739-bf4b-cbf5dff12549?nodeId=delta-stable-2-8s82m-1306261796)
- test steps:
   1. create collection -> index -> insert 5m pk from [0, 5m) -> flush -> index -> load
   2. concurrent: flush + search + upsert (pk start from 0)
  
- pods:
```
delta-stable-dml-2-milvus-datanode-77f477d698-bz2h4               1/1     Running            0                20h     10.104.14.23    4am-node18   <none>           <none>
delta-stable-dml-2-milvus-datanode-77f477d698-rrxvk               1/1     Running            0                20h     10.104.31.179   4am-node34   <none>           <none>
delta-stable-dml-2-milvus-indexnode-f6577cd49-6kmrc               1/1     Running            0                20h     10.104.5.20     4am-node12   <none>           <none>
delta-stable-dml-2-milvus-indexnode-f6577cd49-nnpsw               1/1     Running            0                20h     10.104.4.208    4am-node11   <none>           <none>
delta-stable-dml-2-milvus-mixcoord-f48f47957-24v6z                1/1     Running            0                20h     10.104.1.223    4am-node10   <none>           <none>
delta-stable-dml-2-milvus-proxy-649779845-5fmn7                   1/1     Running            0                20h     10.104.14.26    4am-node18   <none>           <none>
delta-stable-dml-2-milvus-querynode-55bc548f86-2n6g5              1/1     Running            0                20h     10.104.14.25    4am-node18   <none>           <none>
delta-stable-dml-2-milvus-querynode-55bc548f86-bn5jk              1/1     Running            0                20h     10.104.1.224    4am-node10   <none>           <none>
delta-stable-dml-2-milvus-querynode-55bc548f86-d4rm6              1/1     Running            0                20h     10.104.24.241   4am-node29   <none>           <none>
```

**Comment by ThreadDao** [comment]
@XuanYang-cn  zero  task num, but 
The number of Level0DeleteCompaction-executing and zero Level0DeleteCompaction-pending are both 0, but the number of Level0DeleteCompaction-done is greater than 0.
- image: milvus-io-master-c0f0548-20240116
- argo:  [level-zero-stable-9-1877873335](https://argo-workflows.zilliz.cc/archived-workflows/qa/a55efc90-de3e-4ae6-a28b-0820c7529317?nodeId=level-zero-stable-9-1877873335)
- grafana: [metrics of level-zero-par-key-9](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=level-zero-par-key-9&var-collection=All&var-app_name=milvus&from=1705404000000&to=1705449600000)
 - dataCoord compaction task num
 
![image](https://github.com/milvus-io/milvus/assets/27288593/db929576-2736-40eb-ba05-f79471673a46)

- test steps:

create collection with partition_key field and 64 num_partitions -> index -> insert 10m pk from [0, 10m) -> flush -> index -> load
concurrent: query + search + upsert (pk start from 0)

**Comment by XuanYang-cn** [comment]
> @XuanYang-cn Legacy segment state in milvus_datanode_autoflush_buffer_op_count and milvus_datanode_flush_buffer_op_count

@ThreadDao Should be fixed by #30129

**Comment by XuanYang-cn** [comment]
> @XuanYang-cn negative Dropped-L1 segments and wrong Flushed-L1 segments

Should be fixed by #30111

**Comment by XuanYang-cn** [comment]
@ThreadDao Please open other issues when encounters incorrect metrics
